frontend/src/metabase/entities/schemas.js

```diff
@@ -180,3 +193,42 @@ function addTableAvoidingDuplicates(tables, tableId) {
   }
   return tables.includes(tableId) ? tables : [...tables, tableId];
 }
+
+const useGetQuery = ({ id, ...args }) => {
+  const [dbId, schemaName, opts] = parseSchemaId(id);
+
+  if (!dbId || schemaName === undefined) {
+    throw new Error("Schemas ID is of the form dbId:schemaName");
+  }
+
+  const query = { id: dbId, schema: schemaName, ...args };
+
+  const virtualDatabaseTables = useListVirtualDatabaseTablesQuery(
+    opts?.isDatasets ? query : skipToken,
+  );
+
+  const databaseSchemaTables = useListDatabaseSchemaTablesQuery(
+    opts?.isDatasets ? skipToken : query,
+  );
+
+  const tables = opts?.isDatasets
```

Comment:
Conditional call, to mimic this:
https://github.com/metabase/metabase/blob/ba3b7818e6cdf155a74bf00f2b3310c9eec616fc/frontend/src/metabase/entities/schemas.js#L64-L90